google/cloud/internal/openssl_util.cc

```diff
@@ -31,6 +40,118 @@ GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 namespace internal {
 
 namespace {
+#ifdef _WIN32
```

Comment:
Seems like we should 

1) rename `google/cloud/internal/openssl_util.h` to something like `google/cloud/internal/sha256_sign.h`
2) split the implementation into two files `google/cloud/internal/bcrypt/sha256_sign.cc` and `google/cloud/internal/openssl/sha256_sign.cc`

I am not sure if `internal/bcrypt` or something referencing `win32` or `cng` are better ideas.  We should ask @scotthart for suggestions.

Reply:
Renamed to `pem_signing.h` and split to `openssl` and `win32` subdirectories.